### PR TITLE
[Patient Intake Sample] Add the Onboarding > New Patient page

### LINKED ISSUE #4999: [Patient Intake] Form should include other commonly requested items such as
* Address and phone number
* Social security number
* Smoking status
* Allergies 
* Current medications
* Emergency contact 
* Medical history
* Family Medical history

This is a sample of medical history template for reference
image
https://www.med.umich.edu/pdf/Health-History-Questionnaire-Family-Medicine-11-Years-of-Age-and-Older.pdf

### LINKED ISSUE #5015: [Patient Intake Sample] Adjust the intake creation flow
A new tab should be added to direct the user to the patient intake form, which will create the patient.

Reference:

Image

### PR DESCRIPTION
**Pull request by pamella**
## Description

- Resolves #4999
- Resolves #5015 
- Refactor the JSON data files:
   - Move ValueSets to their own file
   - Remove no longer used patient example data
- Refactor the bot code to create the patient resource.

## Resolved Discussions
<details>
<summary><strong>Discussion: QuestionnaireResponse.subject</strong></summary>

Context:

> @rahul1 Since the patient is now being created via the bot handler, we no longer have access to the patient resource to set as the `QuestionnaireResponse.subject` as we did before. This change has complicated the process of displaying the "View Intake Form" action on the patient detail page, as we've lost the search filter based on `subject`. Assigning the response subject within the bot isn't a viable solution because it would trigger an infinite loop—the bot runs whenever a response is created or updated. (If there's a way to configure the bot to run only on creation and not on updates, please let me know.)
> 
> To address this issue temporarily, I've added a questionnaire response list page, ensuring that we don't lose access to the response detail page. However, the previous user experience was superior. If you have any recommendations on how to reinstate the "View Intake Form" action on the patient detail page, please keep me posted.
> 

Resolved at: commit a0b8d05

</details>

## Steps to test

- [ ] Build the bots: `npm run build:bots`
- [ ] Upload all necessary data
- [ ] Go to Onboarding > New Patient
- [ ] Submit the form
- [ ] Check if the patient resource has been created properly
- [ ] Check if the response detail page is working

[Screencast from 12-08-2024 16:51:09.webm](https://github.com/user-attachments/assets/b72db359-1a04-4397-9bfc-4741b3d7d639)

### CODE REVIEW

File: examples/medplum-patient-intake-demo/src/components/PatientActions.tsx
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/components/PatientActions.tsx:1]: This change relates to the _Open discussion_ section of the PR description.

File: examples/medplum-patient-intake-demo/src/pages/IntakeFormPage.tsx [hunk 1]
```diff
@@ -13,20 +13,17 @@ export function IntakeFormPage(): JSX.Element {
...
     (response: QuestionnaireResponse) => {
-      if (!questionnaire || !patient || !profile) {
+      if (!questionnaire || !profile) {
         return;
       }
 
       medplum
         .createResource<QuestionnaireResponse>({
           ...response,
-          subject: createReference(patient),
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/pages/IntakeFormPage.tsx:29]: This change relates to the _Open discussion_ section of the PR description.

File: examples/medplum-patient-intake-demo/src/pages/PatientPage.tsx
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/pages/PatientPage.tsx:1]: This change relates to the _Open discussion_ section of the PR description.

File: examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json
Comment (pamella) [inline examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:1]: The code related to the ValueSets is now in the `examples/medplum-patient-intake-demo/data/core/valuesets.json` file.

File: examples/medplum-patient-intake-demo/data/example/example-patient-data.json
Comment (pamella) [inline examples/medplum-patient-intake-demo/data/example/example-patient-data.json:1]: Removed the example patient data now that the patient is created via bot.

**Comment by reshmakh** [comment]
This looks good to me.  I'll let @rahul1 weigh in as well though.

**Comment by rahul1** [comment]
@pamella , this is a very good observation. I think you discussed this in your last meeting, but the solution would be to [use subscription extensions](https://www.medplum.com/docs/subscriptions/subscription-extensions#subscriptions-for-create-only-events) to trigger the bot only when a new intake form is created. 

You will have to modify the script that builds the bots, as it _generates_ a bundle that includes the `Subscription` resources

File: examples/medplum-patient-intake-demo/data/core/valuesets.json [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        "url": "http://hl7.org/fhir/us/core/ValueSet/us-core-usps-state",
+        "name": "us-core-usps-state",
+        "expansion": {
+          "timestamp": "2024-08-12T00:00:00.000Z",
+          "extension": [
+            {
+              "url": "http://hl7.org/fhir/StructureDefinition/valueset-expansionSource",
+              "valueUri": "http://hl7.org/fhir/us/core/ValueSet/us-core-usps-state"
+            }
+          ],
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/data/core/valuesets.json:70]: @rahul1 This addresses State ValueSet https://github.com/medplum/medplum/pull/5026#issuecomment-2283766417

File: examples/medplum-patient-intake-demo/data/core/valuesets.json [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        "name": "smoking-status",
+        "title": "Smoking Status",
+        "expansion": {
+          "timestamp": "2024-08-12T00:00:00.000Z",
+          "extension": [
+            {
+              "url": "http://hl7.org/fhir/StructureDefinition/valueset-expansionSource",
+              "valueUri": "http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.11.20.9.38"
+            }
+          ],
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/data/core/valuesets.json:163]: @rahul1 This addresses the Smoking Status ValueSet https://github.com/medplum/medplum/pull/5049#discussion_r1713653164

File: examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts [hunk 4]
```diff
@@ -31,55 +30,96 @@ import {
...
   addLanguage(patient, answers['preferred-language']?.valueCoding, true);
 
+  // Create the patient resource
+
+  patient = await medplum.createResource(patient);
+
+  // NOTE: Updating the questionnaire response does not trigger a loop because the bot subscription
+  // is configured for "create"-only event.
+  response.subject = createReference(patient);
+  await medplum.updateResource(response);
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts:121]: This change allows us to keep rendering the "View Intake Form" on the patient detail page.

File: examples/medplum-patient-intake-demo/src/scripts/deploy-bots.ts [hunk 5]
```diff
@@ -61,6 +61,12 @@ async function main(): Promise<void> {
             reason: botName,
             channel: { endpoint: botUrlPlaceholder, type: 'rest-hook' },
             criteria: botDescription.criteria,
+            extension: [
+              {
+                url: 'https://medplum.com/fhir/StructureDefinition/subscription-supported-interaction',
+                valueCode: 'create',
+              },
+            ],
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/scripts/deploy-bots.ts:69]: Relates to https://github.com/medplum/medplum/pull/5059#discussion_r1714303216

File: examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts [hunk 6]
```diff
@@ -314,16 +314,34 @@ export async function addAllergy(
...
+      verificationStatus: {
+        text: 'Unconfirmed',
+        coding: [
+          {
+            system: 'http://hl7.org/fhir/ValueSet/allergyintolerance-verification',
+            code: 'unconfirmed',
+            display: 'Unconfirmed',
+          },
+        ],
+      },
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts:342]: @rahul1 This addresses the allergy statuses 
- https://github.com/medplum/medplum/pull/5033#discussion_r1713645075
- https://github.com/medplum/medplum/pull/5033#discussion_r1713645680

**Comment by pamella** [comment]
@rahul1 I've made the adjustments so the PR is ready for another round of review. I also updated the PR description to reflect the changes.

File: examples/medplum-patient-intake-demo/data/core/valuesets.json [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+            }
+          ]
+        }
+      }
+    },
+    {
+      "fullUrl": "urn:uuid:6abaa6c8-d822-46a4-b3a2-70639475e5ae",
+      "request": {
+        "method": "POST",
+        "url": "ValueSet?name=us-core-usps-state",
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/data/core/valuesets.json:54]:
```suggestion
        "url": "ValueSet",
```
When doing a POST, the query parameters are ignored. 


https://www.medplum.com/docs/fhir-datastore/fhir-batch-requests#conditional-batch-actions

File: examples/medplum-patient-intake-demo/data/core/valuesets.json [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+            { "system": "https://www.usps.com/", "code": "AA", "display": "Armed Forces Americas (except Canada)" }
+          ]
+        }
+      }
+    },
+    {
+      "fullUrl": "urn:uuid:183a569a-7f6d-41f9-97d8-57691fbc2876",
+      "request": {
+        "method": "POST",
+        "url": "ValueSet?name=smoking-status",
```
Comment (rahul1) [inline examples/medplum-patient-intake-demo/data/core/valuesets.json:146]:
```suggestion
        "url": "ValueSet",
```

**Review by rahul1 — APPROVED** [review]

**Pull request merged**